hack/common.sh

```diff
@@ -6,7 +6,7 @@ determine_cri_bin() {
     elif [ "${KUBEVIRT_CRI}" = "docker" ]; then
         echo docker
     else
-        if podman ps >/dev/null 2>&1; then
+        if curl --unix-socket "${XDG_RUNTIME_DIR}/podman/podman.sock" http://d/v3.0.0/libpod/info >/dev/null 2>&1; then
```

Comment:
Checking for the socket here makes the logic for determining whether to use `podman` or `docker` uniform across the `make` targets. Without the PR `make cluster-up` would use `docker` and `make cluster-sync` would use `podman`, if both are installed at the same time and `podman.sock` is not available.

In it's default configuration the `podman.sock` is not available as `podman` needs no daemon by default. It must be explicitly enabled. The PR makes `podman` opt-in from the user's perspective. `docker` always runs a daemon and therefore `docker ps` will not work without an accessible `docker` socket.

I thought of using `podman.sock` as a convenient way to switch between rootless and rootful `podman`. However I just noticed there are instances where `KUBEVIRT_CRI` is called directly. Calling `KUBEVIRT_CRI` directly means always using rootless `podman` unless `make` is called as root. This could be alleviated by passing the `podman.sock` to `podman` (as `--url` argument). What do you think?